lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -1331,8 +1391,767 @@
+             * sets the pattern and all of the relevant options implemented by the pattern on the provided dtf before returning.
+             * 
+             * @param {Object} dtf the DateTimeFormat internal object
+             * @param {Object} options the options object originally given by the user
+             */
+            const getPatternForOptions = (function () {
+                // symbols come from the Unicode LDML: http://www.unicode.org/reports/tr35/tr35-dates.html#Date_Field_Symbol_Table
+                const symbolForOption = {
+                    weekday: "E",
+                    era: "G",
+                    year: "y",
+                    month: "M",
+                    day: "d",
+                    // for hour, we have some special handling
+                    hour: "j", hour12: "h", hour24: "H",
+                    minute: "m",
+                    second: "s",
+                    timeZoneName: "z",
+                };
+                // NOTE - keep this up to date with the map in PlatformAgnostic::Intl::GetDateTimePartKind and the UDateFormatField enum
+                const optionForSymbol = {
+                    E: "weekday", c: "weekday", e: "weekday",
+                    G: "era",
+                    y: "year", u: "year", U: "year",
+                    M: "month", L: "month",
+                    d: "day",
+                    h: "hour", H: "hour", K: "hour", k: "hour",
+                    m: "minute",
+                    s: "second",
+                    z: "timeZoneName", Z: "timeZoneName", v: "timeZoneName", V: "timeZoneName", O: "timeZoneName", X: "timeZoneName", x: "timeZoneName",
+                };
+
+                // lengths here are how many times a symbol is repeated in a skeleton for a given option
+                // the Intl spec recommends that Intl "short" -> CLDR "abbreviated" and Intl "long" -> CLDR "wide"
+                const symbolLengthForOption = {
+                    numeric: 1,
+                    "2-digit": 2,
+                    short: 3,
+                    long: 4,
+                    narrow: 5,
+                };
+                const optionForSymbolLength = {
+                    1: "numeric",
+                    2: "2-digit",
+                    3: "short",
+                    4: "long",
+                    5: "narrow",
+                };
+
+                // for fixing up the hour pattern later
+                const patternForHourCycle = {
+                    h12: "h",
+                    h23: "H",
+                    h11: "K",
+                    h24: "k",
+                };
+                const hourCycleForPattern = {
+                    h: "h12",
+                    H: "h23",
+                    K: "h11",
+                    k: "h24",
+                };
+
+                return function (dtf, options) {
+                    const resolved = _.reduce(dateTimeComponents, function (r, component) {
```

Comment:
>r [](start = 76, length = 1)

rename to accumulator as per [https://lodash.com/docs#reduce](https://lodash.com/docs#reduce)